File: app/src/main/java/com/rdev/tryp/trip/tryp_car/TrypCarAdapter.java
package com.rdev.tryp.trip.tryp_car;

import com.rdev.tryp.firebaseDatabase.model.Driver;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

class TrypCarAdapter extends FragmentStatePagerAdapter {

    private List<?> drivers;

    TrypCarAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        if(drivers.isEmpty() || drivers.size() == 0){
            return null;
        }
        return new TrypCarFragment((Driver) drivers.get(position));
    }

    public void setDrivers(List<?> drivers){
        this.drivers = drivers;
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return drivers.size();
    }

}

File: app/src/main/java/com/rdev/tryp/trip/tryp_car/TrypCarAdapter.kt
package com.rdev.tryp.trip.tryp_car

import com.rdev.tryp.firebaseDatabase.model.Driver
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter


internal class TrypCarAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    private lateinit var drivers: List<*>

    override fun getItem(position: Int): Fragment? {
        return if (drivers.isEmpty()) {
            null
        } else TrypCarFragment(drivers[position] as Driver)
    }

    fun setDrivers(drivers: List<*>) {
        this.drivers = drivers
        notifyDataSetChanged()
    }

    override fun getCount() = drivers.size

}